src/LevelParser.cpp

```diff
@@ -0,0 +1,186 @@
+void parseTileLayer(WorldMap& worldMap, const TiXmlElement & tileLayerElement, const LevelDetails& levelDetails, std::string&& tileLayerName);
+LevelDetails parseLevelDetails(const TiXmlElement& root);
+std::vector<std::vector<int>> decodeTileLayer(const TiXmlElement & tileLayerElement, const LevelDetails & levelDetails);
+const TiXmlElement* findNode(const TiXmlElement& root, const std::string& name);
+const TiXmlElement* findNode(const TiXmlElement& root, const std::string& value, const std::string& name);
+int getNumberOfTileLayers(const TiXmlElement& root);
+int getNumberOfTileSets(const TiXmlElement& root);
+
+void LevelParser::parseLevel(WorldMap& worldMap, const std::string & levelName)
+{
+	//Load XML file
+	TiXmlDocument levelDocument;
+	const bool isOpen = levelDocument.LoadFile(levelName.c_str());
+	assert(isOpen);
+
+	const TiXmlElement& rootNode = *levelDocument.RootElement();
+	const LevelDetails levelDetails = parseLevelDetails(rootNode);
+	parseTileMap(worldMap, rootNode, levelDetails);
+}
+
+void parseTileLayer(WorldMap& worldMap, const TiXmlElement & tileLayerElement, const LevelDetails& levelDetails, std::string&& tileLayerName)
+{
+	worldMap.assignTileLayer(decodeTileLayer(tileLayerElement, levelDetails), levelDetails.m_mapSize, std::move(tileLayerName));
+}
+
+LevelDetails parseLevelDetails(const TiXmlElement & root)
+{
+	int width = 0, height = 0, tileSize = 0;
+	root.Attribute("width", &width);
+	root.Attribute("height", &height);
+	root.Attribute("tilewidth", &tileSize);
+
+	return LevelDetails(tileSize, sf::Vector2i(width, height));
+}
+
+const TiXmlElement * findNode(const TiXmlElement& root, const std::string & name)
+{
+	for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == name)
+		{
+			return e;
+		}
+	}
+	return nullptr;
+}
+
+const TiXmlElement *findNode(const TiXmlElement & root, const std::string & value, const std::string & name)
+{
+	for (const TiXmlElement* e = root.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == value && e->Attribute("name") == name)
+		{
+			return e;
+		}
+	}
+
+	return nullptr;
+}
+
+std::vector<std::vector<int>> decodeTileLayer(const TiXmlElement & tileLayerElement, const LevelDetails & levelDetails)
+{
+	std::vector<std::vector<int>> tileData;
+
+	std::string decodedIDs; //Base64 decoded information
+	const TiXmlElement* dataNode = nullptr; //Store our node once we find it
+	for (const TiXmlElement* e = tileLayerElement.FirstChildElement(); e != nullptr; e = e->NextSiblingElement())
+	{
+		if (e->Value() == std::string("data"))
+		{
+			dataNode = e;
+		}
+	}
+
+	Base64 base64;
+	//Get the text from within the node and use base64 to decode it
+	for (const TiXmlNode* e = dataNode->FirstChild(); e != nullptr; e = e->NextSibling())
+	{
+		const TiXmlText* text = e->ToText();
+		std::string t = text->Value();
+		decodedIDs = base64.base64_decode(t);
+	}
```

Comment:
This either 1) shouldn't loop and use only the first child or 2) append? (need to check Tiled documentation) to decodedIDs instead of overwriting it.